astropy/io/fits/tests/test_diff.py

```diff
@@ -806,9 +806,6 @@ def test_file_output_overwrite_vs_clobber(self):
         diffobj = HeaderDiff(ha, hb)
         diffobj.report(fileobj=outpath)
         report_as_string = diffobj.report()
-        with catch_warnings(AstropyDeprecationWarning) as warning_lines:
+        # Clobber argument is no longer supported
+        with pytest.raises(TypeError):
```

Comment:
I don't think we need to keep the tests that test for a removed argument. I don't quite remember but is the `clobber` warning the only purpose of this test?